src/main/java/com/datatorrent/netlet/DefaultEventLoop.java

```diff
@@ -323,7 +323,12 @@ public void run()
             }
           }
           else {
-            register(channel, SelectionKey.OP_CONNECT, l);
+            /*
```

Comment:
I guess it fires OP_CONNECT or OP_CONNECT and OP_READ as adding OP_READ fixes the problem and OP_CONNECT is removed from interestOps.

I ran few dozens tests where ClientListener.read() checks that ClientListener.connected() was called and that was always the case. It is hard to say whether during those tests the original issue was triggered or not. Usually it is triggered once in 10-20 runs, but adding more logging changes how often it reproduces.